lib/npc-generation/fetchProfessionChance.test.ts

```diff
@@ -0,0 +1,154 @@
+import { fetchProfessionChance } from './fetchProfessionChance'
+import { setRandomFloat } from '../src/randomFloat'
+import { Town, TownProfessions } from '../town/_common'
+import { NPC } from './_common'
+
+beforeAll(() => {
+  setRandomFloat(() => 0.5)
+})
+
+const getProfession = (args = {}) => ({
+  professionType: 'profession',
+  professionSector: 'space',
+  domSub: 'sub',
+  socialClass: 'low',
+  population: 1,
+  ...args
+})
+
+const getProfessions = (args = {}) => ({
+  astronaut: { ...getProfession({ population: 5 }) },
+  meadarie: { ...getProfession({ sector: 'booze', type: 'bartender', domSub: 'dom', socialClass: 'high' }) },
+  ...args
+})
+
+const getTownDefaults = (args = {}) => ({
+  professions: { ...getProfessions() },
```

Comment:
No need to spread these if we're not injecting values.
```suggestion
  professions: getProfessions(),
```

Reply:
The reason I'm using the spread operator here is because the spread operator does a [shallow clone](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals) for objects. That means `getTown().professions.astronaut` without the spread operator is passed by reference to all the tests, and the tests for gender norms explicitly mutates them, so that it affects the later tests. This was causing later tests to fail, hence the inclusion.

Usually I solve this by using something like lodash's [clone deep](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals), which would dereference all the objects, but I didn't want to add yet another dependency here.

Reply:
I know, but in those cases it's usually a better choice to create a new object immutably in that particular test. That way you don't rely on the referential stability of the objects you're using.

For example, this will still mutate the original object, if you're spreading professions.
```js
townProps.professions.astronaut.exclusions = mockFn
```

The best case would be that `getTownDefaults` would return a new object when called, so you can just instantiate a new one in every test case . Not sure if it does that already, since It's been a while since I worked on this code.

Either way, it's not a blocking issue.